### PR TITLE
refactor: Nested search params in ES config

### PR DESCRIPTION
**Pull request by KShivendu**
@filipecosta90 I realised that the problem with `parallel` that you highlighted in your PR https://github.com/qdrant/vector-db-benchmark/pull/103 can be solved in a simpler way like this. We do the same with other engines. 

I've added you a co-author since parts of the code come from your PR :)

### CODE REVIEW

**Comment by filipecosta90** [comment]
LGTM, thanks for pinging me on this :+1:

**Review by filipecosta90 — APPROVED** [review]

File: engine/clients/elasticsearch/config.py [hunk 1]
```diff
@@ -1,4 +1,30 @@
...
+from elasticsearch import Elasticsearch
+
+ELASTIC_PORT = int(os.getenv("ELASTIC_PORT", 9200))
+ELASTIC_INDEX = os.getenv("ELASTIC_INDEX", "bench")
+ELASTIC_USER = os.getenv("ELASTIC_USER", "elastic")
+ELASTIC_PASSWORD = os.getenv("ELASTIC_PASSWORD", "passwd")
+
+ELASTIC_TIMEOUT = int(os.getenv("ELASTIC_TIMEOUT", 300))
+ELASTIC_INDEX_TIMEOUT = os.getenv("ELASTIC_INDEX_TIMEOUT", "30m")
+ELASTIC_INDEX_REFRESH_INTERVAL = os.getenv("ELASTIC_INDEX_REFRESH_INTERVAL", "-1")
```
Comment (joein) [inline engine/clients/elasticsearch/config.py:12]: there are connection parameters, aren't they?
why are we setting them with environment variables?
should not they be in config?

File: engine/clients/elasticsearch/config.py [hunk 2]
```diff
@@ -1,4 +1,30 @@
...
+
+
+def get_es_client(host, connection_params):
+    client: Elasticsearch = None
```
Comment (joein) [inline engine/clients/elasticsearch/config.py:16]: no need to set it `=None`

File: engine/clients/elasticsearch/config.py (hunk 1 above)
Comment (joein) [inline engine/clients/elasticsearch/config.py:12]: These 2 params seems to be unused

Comment (KShivendu) [reply engine/clients/elasticsearch/config.py]: Good point. Realised we don't need any of them. So removed all 3.

File: engine/clients/elasticsearch/config.py (hunk 2 above)
Comment (KShivendu) [reply engine/clients/elasticsearch/config.py]: Done. Thanks!

File: engine/clients/elasticsearch/config.py (hunk 1 above)
Comment (KShivendu) [reply engine/clients/elasticsearch/config.py]: Removed

File: engine/clients/elasticsearch/configure.py [hunk 3]
```diff
@@ -18,26 +13,15 @@ class ElasticConfigurator(BaseConfigurator):
         Distance.COSINE: "cosine",
         Distance.DOT: "dot_product",
     }
+    # TODO: Add other types
     INDEX_TYPE_MAPPING = {
```
Comment (joein) [inline engine/clients/elasticsearch/configure.py:17]: which types we don't support at the moment?

Comment (KShivendu) [reply engine/clients/elasticsearch/configure.py:17]: `keyword, text, float`. Added them right away.

File: engine/clients/elasticsearch/configure.py [hunk 4]
```diff
@@ -72,12 +56,8 @@ def recreate(self, dataset: Dataset, collection_params):
...
                         "similarity": self.DISTANCE_MAPPING[dataset.config.distance],
                         "index_options": {
-                            **{
-                                "type": "hnsw",
-                                "m": 16,
-                                "ef_construction": 100,
-                            },
-                            **collection_params.get("index_options"),
+                            "type": "hnsw",
+                            **collection_params["index_options"],
```
[the remark this answers is not in the document]
Comment (KShivendu) [reply engine/clients/elasticsearch/configure.py:60]: > should not there be some similar changes in opensearch as well?

We actually just did that in https://github.com/qdrant/vector-db-benchmark/pull/122

[the remark this answers is not in the document]
Comment (joein) [reply engine/clients/elasticsearch/configure.py:62]: and #122 is just a disaster in terms of backwards compatibility

[the remark this answers is not in the document]
Comment (KShivendu) [reply engine/clients/elasticsearch/configure.py:62]: We agreed that this is okay.

**Review by generall — APPROVED** [review]

**Pull request merged**